ggml.h

```diff
@@ -1044,6 +1044,24 @@ extern "C" {
             int                   n_dims,
             int                   mode);
 
+    // same as ggml_rope but allows specifying p scale factor
+    GGML_API struct ggml_tensor * ggml_rope_scaled(
+            struct ggml_context * ctx,
+            struct ggml_tensor  * a,
+            int                   n_past,
+            int                   n_dims,
+            int                   mode,
+            float                 p_scale);
+
+    // same as ggml_rope_inplace but allows specifying p scale factor
+    GGML_API struct ggml_tensor * ggml_rope_scaled_inplace(
+            struct ggml_context * ctx,
+            struct ggml_tensor  * a,
+            int                   n_past,
+            int                   n_dims,
+            int                   mode,
+            float                 p_scale);
```

Comment:
No need to extend the API - add `p_scale` to original `ggml_rope_xxx()` and add comment to use `p_scale == 1.0f` for regular computation. Add `GGML_ASSERT(p_scale == 1.0f)` in backward call

Reply:
> No need to extend the API - add `p_scale` to original `ggml_rope_xxx()`

Won't this break every single thing that currently uses the llama.cpp version of GGML?

What do you think about using a define to enable the p_scale argument for rope and having it be off by default? That way existing stuff can opt in.

It might also be worth thinking about adding something like `GGML_API_VERSION` which could be bumped when incompatible changes occur, so stuff building against GGML could handle API changes more gracefully.